Added Data Theorem step doc (#411)
appcircleio/appcircle-docusaurus: docs/workflows/common-workflow-steps/data-theorem-mobile-secure.md

---
title: Data Theorem Mobile Secure
description: Data Theorem Mobile Secure, enables users to automatically scan their mobile applications for security vulnerabilities and compliance issues directly from the Appcircle.
tags: [data-theorem, secure, mobile]
sidebar_position: 8
---

import Screenshot from '@site/src/components/Screenshot';

# Data Theorem Mobile Secure

> The **Data Theorem Mobile Secure** step on Appcircle integrates the [Data Theorem Mobile Secure](https://www.datatheorem.com/products/mobile-secure/) service into the CI/CD workflow. It enables users to automatically scan their mobile applications for security vulnerabilities and compliance issues directly from the Appcircle environment. This step facilitates proactive security testing, allowing developers to identify and resolve potential security threats before deploying their mobile applications.

Reviewer: ```suggestion
The **Data Theorem Mobile Secure** step integrates the [Data Theorem Mobile Secure](https://www.datatheorem.com/products/mobile-secure/) service into the CI/CD workflow on Appcircle. This step allows users to automatically scan their mobile applications for security vulnerabilities and compliance issues, facilitating proactive security testing. Developers can identify and resolve potential security threats before deploying their mobile applications.
```

### Prerequisites

> The workflow steps that need to be executed before running the **Data Theorem Mobile Secure** workflow step, along with their respective reasons, are listed in the table below. Prerequisites vary depending on the platform used:
> 

Reviewer: ```suggestion
The workflow steps that need to be executed before running the **Data Theorem Mobile Secure** step vary depending on the platform and are listed below:
```

<Screenshot url='https://cdn.appcircle.io/docs/assets/common-workflow-components-datatheorem_3.png'/>

#### For iOS Flutter

| Prerequisite Workflow Step        | Description                                                                                              |
| --------------------------------- | -------------------------------------------------------------------------------------------------------- |
| [**Xcodebuild for Devices**](https://docs.appcircle.io/workflows/ios-specific-workflow-steps#xcodebuild-for-devices-archive--export) | Builds the application in ARM architecture and generates an `IPA` file. |
| [**Flutter Build for iOS**](https://docs.appcircle.io/workflows/flutter-specific-workflow-steps#flutter-build-for-ios) | Prepares the Flutter project for the iOS environment and builds it using the [Flutter SDK](https://github.com/flutter/flutter). |

<Screenshot url='https://cdn.appcircle.io/docs/assets/common-workflow-components-datatheorem_4.png'/>

### Input Variables

> For each component, specific input variables are required for its operation on your system. The input variables necessary for **Data Theorem Mobile Secure** are as follows:
> 

Reviewer: ```suggestion
Specific input variables are required for the operation of each component in your system. The following are the input variables necessary for the **Data Theorem Mobile Secure** step:
```
